core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -164,6 +174,9 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
           driverConf.set(key, value)
         }
       }
+      // Periodically update the credentials for this user to ensure HDFS tokens get updated.
+      val tokenUpdater = new ExecutorDelegationTokenUpdater(driverConf, SparkHadoopUtil.get.conf)
+      tokenUpdater.updateCredentialsIfRequired()
```

Comment:
It won't cause it to go check immediately. This call basically starts the updater, which will check for the tokens periodically (at 80% of expiry). I can rename this to `start` if it makes the intent clearer.